src/LINEBot.php

```diff
@@ -878,21 +879,51 @@ public function getNarrowcastProgress($requestId)
      */
     public function createAudienceGroupForUpdatingUserIds(
         $description,
-        $audiences,
+        $audiences = [],
         $isIfaAudience = false,
         $uploadDescription = null
     ) {
         $params = [
             'description' => $description,
             'isIfaAudience' => $isIfaAudience,
-            'audiences' => $audiences,
         ];
+        if (!empty($audiences)) {
+            $params['audiences'] = $audiences;
+        }
         if (isset($uploadDescription)) {
             $params['uploadDescription'] = $uploadDescription;
         }
         return $this->httpClient->post($this->endpointBase . '/v2/bot/audienceGroup/upload', $params);
     }
 
+    /**
+     * Create audience for uploading user IDs (by file)
+     *
+     * @param string $description The audience's name. Max character limit: 120
+     * @param string $filePath A text file path with one user ID or IFA entered per line. Max number: 1,500,000
+     * @param bool $isIfaAudience If this is false (default), recipients are specified by user IDs.
+     * @param string|null $uploadDescription The description to register with the job.
+     * @return Response
+     */
+    public function createAudienceGroupForUploadingUserIdsByFile(
+        $description,
+        $filePath,
+        $isIfaAudience = false,
+        $uploadDescription = null
+    ) {
+        $params = [
+            'description' => $description,
+            'isIfaAudience' => $isIfaAudience,
+            'file' => new CURLFile($filePath, 'text/plain', 'file'),
```

Comment:
I think LINE Bot should not know CURL. But I could not think of a good idea. So I directly specify CURLFile.